### ISSUE TITLE
Bound box detection is is too big

### ISSUE THREAD

**Issue by AweSwan** [AlexeyAB/darknet#526]
When yolo detects an object it's bounding box takes up about 95% of the screen.

![image](https://user-images.githubusercontent.com/21250980/38075016-2a9bd2f8-3374-11e8-9b61-66154a158d2a.png)

My data-set is generated from renders of a animated greyhound in most angles that would be seen by a camera. eg.
![image](https://user-images.githubusercontent.com/21250980/38075114-8f96ed46-3374-11e8-93d0-f0a424f17601.png)

I suspect it is because every images has the same bounding box eg 
annotation file are automatically generated to be:
0 0.500000 0.500000 1.000000 1.000000 <- which is the entire image.

I'm running a test to see if this is the case.

Some other things it might be:
If yolo discovers old .weights file in the backup folder does it completely overwrite the the old files or does it use those weights again ?

Here is my config file: ( I only changed the classes to 1 and filter to 30)
[net]
batch=64
subdivisions=8
width=416
height=416
channels=3
momentum=0.9
decay=0.0005
angle=0
saturation = 1.5
exposure = 1.5
hue=.1

learning_rate=0.001
max_batches = 40200
policy=steps
steps=-1,100,20000,30000
scales=.1,10,.1,.1

[convolutional]
batch_normalize=1
filters=16
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=32
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=64
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=128
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=256
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=512
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=1

[convolutional]
batch_normalize=1
filters=1024
size=3
stride=1
pad=1
activation=leaky

###########

[convolutional]
batch_normalize=1
size=3
stride=1
pad=1
filters=1024
activation=leaky

[convolutional]
size=1
stride=1
pad=1
filters=30
activation=linear

[region]
anchors = 1.08,1.19,  3.42,4.41,  6.63,11.38,  9.42,5.11,  16.62,10.52
bias_match=1
classes=1
coords=4
num=5
softmax=1
jitter=.2
rescore=1

object_scale=5
noobject_scale=1
class_scale=1
coord_scale=1

absolute=1
thresh = .6
random=1

**Comment by AweSwan** [comment]
Ok, 

So I changed the annotation file bounding box to be about 0.25 of the screen size eg ..
0 0.500000 0.500000 0.500000 0.500000
and it detects bounding box it always 0,25 of the screen.
![image](https://user-images.githubusercontent.com/21250980/38075753-d0bcdba8-3376-11e8-9583-ce773bb7163a.png)

So this seems to be the issue.

Maybe I could use these variable to get different bounding box scales ? ( any other suggestion welcome!)

object_scale=5
noobject_scale=1
class_scale=1
coord_scale=1

**Comment by AlexeyAB** [comment]
* You should check all your images using this tool: https://github.com/AlexeyAB/Yolo_mark

* Each horse should be labeled as separate object.

**Comment by RushNuts** [comment]
@AweSwan Hi! Can you tell more about your dataset? Do you use images where each contains one object and selects the entire image? What's size your images? Do you before train with this dataset?

**Comment by AweSwan** [comment]
@AlexeyAB 
Thanks  for your response! 
I did check the images with Yolo_Mark. Originally every image was full screen. It looks like yolo does not do much scaling. Each greyhound was marked as a separately.

@RushNuts 
The entire dataset was generated from a 3D renders (in unity3D) of an animating greyhound. The entire image was selected. There was always only one object in the scene. The resolution changes as the images were always cropped, but were about 400x400.

I think the fix will be to not crop the images and have the dogs rendered at roughly the same size ratio that they appear in the video.

I am not sure if yolo takes into consideration the number of dogs in the training. This will always be 1.

**Comment by AlexeyAB** [comment]
@AweSwan 

desirable that your training dataset include images with objects at **diffrent: scales**, rotations, lightings, from different sides, **on different backgrounds**

1. If your objects always full-screen, then network will detect objects only full screen (will not detect objects at different scales) - you can try to fix it by increasing data augmentation - set `jitter=.9` https://github.com/AlexeyAB/darknet/blob/1b85111217e90bc20907070283d5de9d9d750057/cfg/yolov3.cfg#L785

2. If your objects always on a homogeneous background, then network will detect objects only a homogeneous background (will not detect on real non-homogeneous backgrounds)


But it is very desirable that the training dataset contains objects of different sizes and in different places of the image, as well as on a various backgrounds. 
Or at least you should add images with backgrounds but without Dogs (without labels = empty txt-files) - negative samples: https://github.com/AlexeyAB/darknet#how-to-improve-object-detection

> desirable that your training dataset include images with non-labeled objects that you do not want to detect - negative samples without bounded box

**Comment by AweSwan** [comment]
@AlexeyAB Thanks for the great information! It looks like generating images of dogs might not be a good idea. A jitter value above 0.5 was causing darknet to crash. Running a test now with jitter = 0.5. I'm training it with real dogs marked and generated dogs images. I am hoping yolo might be able to distinguish *between* the dog numbers. It already does a great job detecting generic greyhounds with manually marked greyhounds.

**Comment by AlexeyAB** [comment]
The idea with the generation of 3d images is interesting. Thus, you can get a lot of images at different scales, rotations, sides, and lighting.

**Comment by RushNuts** [comment]
@AlexeyAB 
Hello! I would like to clarify on the same issue for the preparation of the data. At me here there are images 256х256 (generation of images the sizes from 125х125 to 500х500) in which the object almost on all image is represented. The background of the images is different, but the images are the same and entirely on the whole image. Images are marked entirely. With such a set of output, I can not train correctly? Comment please

@AweSwan Okay, thanks

**Comment by AlexeyAB** [comment]
@RushNuts This is not the best option for training. You can try to train with `jitter=0.5` or better `jitter=0.9` and check what mAP can your get.
